packages/console/src/pages/UserDetails/UserSettings/UserMfaVerifications/index.tsx

```diff
@@ -0,0 +1,99 @@
+import { type UserMfaVerificationResponse } from '@logto/schemas';
+import { Trans, useTranslation } from 'react-i18next';
+import useSWR from 'swr';
+
+import MfaFactorName from '@/components/MfaFactorName';
+import MfaFactorTitle from '@/components/MfaFactorTitle';
+import Button from '@/ds-components/Button';
+import Table from '@/ds-components/Table';
+import useApi, { type RequestError } from '@/hooks/use-api';
+import { useConfirmModal } from '@/hooks/use-confirm-modal';
+import { type UserMfaVerification } from '@/types/mfa';
+
+import * as styles from './index.module.scss';
+
+type Props = {
+  userId: string;
+};
+
+function UserMfaVerifications({ userId }: Props) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console.user_details.mfa' });
+  const {
+    data: mfaVerifications,
+    error,
+    isLoading,
+    mutate,
+  } = useSWR<UserMfaVerificationResponse, RequestError>(`api/users/${userId}/mfa-verifications`);
+
+  const api = useApi();
+  const { show: showConfirm } = useConfirmModal();
+
+  const handleDelete = async (mfaVerification: UserMfaVerification) => {
+    const [result] = await showConfirm({
+      ModalContent: () => (
+        <Trans
+          t={t}
+          i18nKey="deletion_confirmation"
+          components={{
+            name: <MfaFactorName {...mfaVerification} />,
+          }}
+        />
+      ),
+      confirmButtonText: 'general.remove',
+    });
+
+    if (!result) {
+      return;
+    }
+
+    await api.delete(`api/users/${userId}/mfa-verifications/${mfaVerification.id}`);
+    void mutate(mfaVerifications?.filter((item) => item.id !== mfaVerification.id));
+  };
+
+  return (
+    <>
+      {!isLoading && !error && (
```

Comment:
will this cause page jump while missing elements?

Reply:
No, `useSWR` will not trigger loading state when revalidating existing data.